From 922bb864713b0980f486ed715482de20e0d59646 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 01 Mar 2021 14:36:17 +0200
Subject: [PATCH] [SW-34388] secureboot: support for preboot images

PPboot brings up the preboot after verifying them for
RSA3072 using arcboot functions. Hence add support
for this flow.

Change-Id: I8678f4777e5c03f770871e7f7e02d95e418a266a
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index b8196f3..f304fbc 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -365,6 +365,9 @@
 	# Create erom binaries
 	prep_image_erom
 
+	# Preboot Image - PRI
+	hl_pack_bin ${ARCBOOT_BINFILE} 64K ${PRIME_IMAGE_RAM_KEY} ${PRIME_IMAGE_RAM_CRC_ERROR} ${PRIME_IMAGE_RAM_SIGN_ERROR}
+
 	# trunk to round 128K as DMA spi can copy chunks of 1KB only
 	truncate -s 128K ${DEV_NAME}-spi-flash.bin
 
@@ -385,6 +388,9 @@
 
 	# update SECONDARY image offset in start 64k padded area
 	hl_img_offset ${DEV_NAME}-spi-flash.bin ${SECOND_IMAGE}
+
+	# Preboot Image - SEC
+	hl_pack_bin ${ARCBOOT_BINFILE} 64K ${SECOND_IMAGE_RAM_KEY} ${SECOND_IMAGE_RAM_CRC_ERROR} ${SECOND_IMAGE_RAM_SIGN_ERROR}
 	truncate -s 256K ${DEV_NAME}-spi-flash.bin
 
 	# To be populated into FW SRAM
